fed/api.py

```diff
@@ -347,21 +347,31 @@ def get(
                 if party_name == current_party:
                     continue
                 else:
-                    send(
-                        party_name,
-                        ray_object_ref,
-                        fed_object.get_fed_task_id(),
-                        fake_fed_task_id,
-                        party_name,
-                    )
+                    if fed_object._was_sending_or_sent_to_party(party_name):
+                        # This object was sending or sent to the target party,
+                        # so no need to do it again.
+                        continue
+                    else:
+                        fed_object._mark_is_sending_to_party(party_name)
+                        send(
+                            party_name,
+                            ray_object_ref,
+                            fed_object.get_fed_task_id(),
+                            fake_fed_task_id,
+                            party_name,
+                        )
```

Comment:
It seems this logic is repeated and could be refactored to a wrapper function, both on the sending and receiving side

Reply:
Yes, goooood catch!
I'm not sure if you're interested in contributing it?